src/app/component/layout/components/header/header.component.ts

```diff
@@ -147,6 +146,11 @@ export class HeaderComponent implements OnInit, OnDestroy {
     this.arrayLang[0].lang = language;
     this.arrayLang[index].lang = temporary;
     this.langDropdownVisible = false;
+    if (this.isLoggedIn) {
+      const curLangId = this.languageService.getLanguageId(language.toLowerCase() as Language);
+      this.userService.updateUserLanguage(curLangId)
```

Comment:
if you don't care what the answer is from BE, you can switch it to a promise and solve it in the service.